src/world/common/UnkNpcDurationFlagFunc.inc.c

```diff
@@ -0,0 +1,16 @@
+#include "common.h"
+#include "map.h"
+
+static void UnkNpcDurationFlagFunc(ScriptInstance* script) {
+    Enemy* enemy = script->owner1.enemy;
+    Npc* npc = get_npc_unsafe(enemy->npcID);
+
+    npc->duration--;
+    if ((npc->duration << 0x10) == 0) {
```

Comment:
iirc, this is actually something like & 0xFFFF or so? Not sure, but this at the very least seems incorrect. (also, if possible shift amounts in decimal. Makes it way easier to recognize patterns)